app/presenters/hyrax/work_show_presenter.rb

```diff
@@ -204,6 +204,13 @@ def manifest_metadata
       metadata
     end
 
+    # determine if the user can add this work to a collection
+    # @return true if the user can deposit to at least one collection OR if the user can create a collection; otherwise, false
+    def can_add_to_collection?(depositable_collections)
```

Comment:
I'm slightly uncomfortable with the `Presenter` leaking this permission information into the view, and the view conditional on https://github.com/samvera/hyrax/pull/3338/files#diff-600b68f4bb5cd3833e5a8be24f8b356eR25 that results.

Could we explore refactors that avoid this--hopefully by making the new presenter method return some presentable data?

One (maybe too awkward?) possibility that comes to mind is to change the new method to change the new method to:

```ruby
  def depositable_work_ids(collections)
    return [] unless collections.present? || current_ability.can?(:create_any, Collection)
    [id]
  end
```

Another option might be to just rename the existing method to something presentation oriented:

```ruby
  def show_deposit_for(collections:)
    collections.present? || current_ability.can?(:create_any, Collection)
  end
```

The second option doesn't squash the need for the view conditional, but does at least reframe so the presenter is focused on responding to messages in its own domain.

Thoughts?